modules/gdnative/pluginscript/pluginscript_language.cpp

```diff
@@ -0,0 +1,407 @@
+	}
+	return -1;
+}
+
+String PluginScriptLanguage::make_function(const String &p_class, const String &p_name, const PoolStringArray &p_args) const {
+	if (this->_desc.make_function) {
+		godot_string tmp = this->_desc.make_function((godot_string *)&p_class, (godot_string *)&p_name, (godot_pool_string_array *)&p_args);
+		String ret = *(String *)&tmp;
+		godot_string_destroy(&tmp);
+		return ret;
+	}
+	return String();
+}
+
+Error PluginScriptLanguage::complete_code(const String &p_code, const String &p_base_path, Object *p_owner, List<String> *r_options, bool &r_force, String &r_call_hint) {
+	if (this->_desc.complete_code) {
+		Array options;
+		godot_error tmp = this->_desc.complete_code(
+				(godot_string *)&p_code,
+				(godot_string *)&p_base_path,
+				(godot_object *)p_owner,
+				(godot_array *)&options,
+				&r_force,
+				(godot_string *)&r_call_hint);
+		for (int i = 0; i < options.size(); i++) {
+			r_options->push_back(String(options[i]));
+		}
+		Error err = *(Error *)tmp;
+		return err;
+	}
+	return ERR_UNAVAILABLE;
+}
+
+void PluginScriptLanguage::auto_indent_code(String &p_code, int p_from_line, int p_to_line) const {
+	if (this->_desc.auto_indent_code) {
+		this->_desc.auto_indent_code((godot_string *)&p_code, p_from_line, p_to_line);
+	}
+	return;
+}
+
+void PluginScriptLanguage::add_global_constant(const StringName &p_variable, const Variant &p_value) {
+	const String variable = String(p_variable);
+	this->_desc.add_global_constant((godot_string *)&variable, (godot_variant *)&p_value);
+}
+
+/* LOADER FUNCTIONS */
+
+void PluginScriptLanguage::get_recognized_extensions(List<String> *p_extensions) const {
+	for (int i = 0; this->_desc.recognized_extensions[i]; ++i) {
+		p_extensions->push_back(String(this->_desc.recognized_extensions[i]));
+	}
+}
+
+void PluginScriptLanguage::get_public_functions(List<MethodInfo> *p_functions) const {
+	// TODO: provid this statically in `godot_pluginscript_language_desc` ?
+	if (this->_desc.get_public_functions) {
+		Array functions;
+		this->_desc.get_public_functions((godot_array *)&functions);
+		for (int i = 0; i < functions.size(); i++) {
+			MethodInfo mi = MethodInfo::from_dict(functions[i]);
+			p_functions->push_back(mi);
+		}
+	}
+}
+
+void PluginScriptLanguage::get_public_constants(List<Pair<String, Variant> > *p_constants) const {
+	// TODO: provid this statically in `godot_pluginscript_language_desc` ?
+	if (this->_desc.get_public_constants) {
+		Dictionary constants;
+		this->_desc.get_public_constants((godot_dictionary *)&constants);
+		for (const Variant *key = constants.next(); key; key = constants.next(key)) {
+			Variant value = constants[key];
+			p_constants->push_back(Pair<String, Variant>(*key, value));
+		}
+	}
+}
+
+void PluginScriptLanguage::profiling_start() {
+#ifdef DEBUG_ENABLED
+	this->lock();
+	if (this->_desc.profiling_start) {
+		this->_desc.profiling_start();
+	}
+	this->unlock();
```

Comment:
I would move the locking inside the if. If the scripting language doesn't provide profiling methods then why even bother locking? There won't be any changes anyway.

Reply:
done.